### ISSUE TITLE
CurseForge external downloads not working properly

### ISSUE THREAD

**Issue by whitebear60** [ATLauncher/ATLauncher#705]
**Describe the bug**
The browser download of the CurseForge mods doesn't handled out properly. More info at the reproduction steps

**To Reproduce**

1. Try to download the mod which is distributed via CurseForge and has third-party downloads disabled (for example, https://www.curseforge.com/minecraft/mc-mods/macos-input-fixes). The browser window should open with the download prompt
2. Save the mod file to the Downloads folder
3. The launcher begins to behave wierdly, constantly opening new browser windows with the prompt of downloading the mod
4. Cancel the download in the launcher, then try to add this mod again
5. Now the launcher creates the empty jar file with 0kb size where the mod should be copied to

**Expected behavior**
On the first download the mod should be copied to instance mods folder normally (or moved, idk how it is handled in the launcher)

**Logs**
[View the logs here](https://paste.atlauncher.com/view/78fa7432-09ee-4560-a22a-2c583fe96f62)

**Screenshots**
I can provide the video demos, if you wish. If so, write in the comments.

**System information (please complete the following information, or remove if not applicable):**

- OS (including distro/version): macOS Ventura 13.0.1
- Java Version: Microsoft OpenJDK 17.0.5

**Comment by whitebear60** [comment]
although the logs are taken from the development version built by myself which is not up-to-date with upstream probably, the bug also happened in 3.4.24.0...

**Comment by whitebear60** [comment]
0kb files are also appearing sometimes when using Add mod feature, although I haven't investigated the exact info when it happens in that case yet...

**Comment by RyanTheAllmighty** [comment]
Yeah this is part of the sandboxing stuff Max started a while back, needs #559 and #502 to be sorted then it likely will work fine as it's likely to be a bundling issue

**Comment by whitebear60** [comment]
This happens on jar version too
